### PR TITLE
PD-1885: segmented control hijack

### PR DESCRIPTION
**Pull request by shaneeza**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [PD-1885](https://jira.mongodb.org/browse/PD-1885)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

File: .changeset/sour-pillows-talk.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/segmented-control': minor
```
Comment (TheSonOfThomp) [inline .changeset/sour-pillows-talk.md:2]: I'd call this a patch

File: packages/segmented-control/src/SegmentedControl.spec.tsx [hunk 2]
```diff
@@ -28,8 +28,6 @@ const renderNewContainer = () => {
   const banana = getByText(rendered.container, 'Banana')
     .parentElement as Element;
 
-  userEvent.tab();
```
Comment (TheSonOfThomp) [inline packages/segmented-control/src/SegmentedControl.spec.tsx:31]: It this isn't needed, that's fantastic!

File: packages/segmented-control/src/SegmentedControlOption.tsx [hunk 3]
```diff
@@ -313,8 +314,12 @@ const SegmentedControlOption = React.forwardRef<
     const didComponentMount = useRef(false);
     const buttonRef = useRef<HTMLButtonElement>(null);
     useEffect(() => {
+      // Check if the component did mount
       if (didComponentMount.current) {
-        if (usingKeyboard && focused) {
+        // usingKeyboard: Returns if the keyboard is being used.
+        // focused: Returns if this option should be the item in focus.
+        // isfocusInComponent: Returns if the focus should organically be this component. Without this check focus will be hijacked to this component if `usingKeyboard` is updated to true.
+        if (usingKeyboard && focused && isfocusInComponent) {
```
Comment (TheSonOfThomp) [inline packages/segmented-control/src/SegmentedControlOption.tsx:322]: 🤘

File: packages/segmented-control/src/SegmentedControl.tsx [hunk 4]
```diff
@@ -295,8 +295,12 @@ const SegmentedControl = React.forwardRef<
   }: SegmentedControlProps,
   forwardedRef,
 ) {
-  // TODO log warning if defaultValue is set but does not match any child value
+  // TODO: log warning if defaultValue is set but does not match any child value
   const { usingKeyboard } = useUsingKeyboardContext();
+  const [segmentedContainer, setSegmentedContainer] = useState<null | Element>(
```
Comment (TheSonOfThomp) [inline packages/segmented-control/src/SegmentedControl.tsx:300]: 🤷 Why use this pattern, vs `useRef`?

**Review by TheSonOfThomp — APPROVED** [review]
Looks good! Just a few small comments, but nothing to hold it back

File: packages/segmented-control/src/SegmentedControl.spec.tsx (hunk 2 above)
Comment (shaneeza) [reply packages/segmented-control/src/SegmentedControl.spec.tsx:31]: Yeah, it took me a while this was the reason the tests were failing. It was making the first item focused but the first test is checking if the first item is not in focus.

File: packages/segmented-control/src/SegmentedControl.tsx (hunk 4 above)
Comment (shaneeza) [reply packages/segmented-control/src/SegmentedControl.tsx:300]: Good question! Turns out `useRef` works fine.

**Pull request merged**